src/Silex/Application.php

```diff
@@ -454,8 +454,8 @@ public function mount($prefix, $controllers)
             }
 
             $controllers = $connectedControllers;
-        } elseif (!$controllers instanceof ControllerCollection) {
-            throw new \LogicException('The "mount" method takes either a "ControllerCollection" or a "ControllerProviderInterface" instance.');
+        } elseif (!$controllers instanceof ControllerCollection && !is_callable($controllers)) {
+            throw new \LogicException('The "mount" method takes either a "ControllerCollection" instance, "ControllerProviderInterface" instance, or a callable.');
```

Comment:
@SpacePossum, I was trying to keep consistent with old error, and instances implement interfaces. I prefer the way the error currently is, but I can change if that's consensus. Either way, this error message is a minor part of the whole PR, I'm more interested in seeing if this design change should be accepted or not, then we can work on the little things.